bitcoin/Cargo.toml

```diff
@@ -40,13 +40,14 @@ bitcoin-internals = { path = "../internals" }
 bech32 = { version = "0.9.0", default-features = false }
 bitcoin_hashes = { version = "0.11.0", default-features = false }
 secp256k1 = { version = "0.25.0", default-features = false, features = ["bitcoin_hashes"] }
-core2 = { version = "0.3.0", optional = true, default-features = false }
 
 base64 = { version = "0.13.0", optional = true }
 bitcoinconsensus = { version = "0.20.2-0.5.0", optional = true, default-features = false }
+# You probably want to use "no-std" instead of using these two features directly.
```

Comment:
I'd actually love to split these and remove `no-std` :P

Reply:
hmmm, I'll keep that in mind. I don't like how we have different crates using std/alloc differently, its a pain to remember (reffering to the fact that sometimes "std" enables "alloc" and sometimes not).

Reply:
I think I already said `std` should always enable `alloc`. `core2` should be orthogonal. Although we need `core2` because of IO traits being everywhere.

Reply:
Yeah, I finally worked this out. I was getting confused by the `HexWriter` that uses `core2` and `alloc` - now its removed and I've started to separate "alloc" from "core2" (both in mind and in PRs).